### ISSUE TITLE
organizeDeclarations bug

### ISSUE THREAD

**Issue by cameroncooke** [nicklockwood/SwiftFormat#1907]
I'm using SwiftFormat with Cursor so that formatting is applied on save and this revealed an issue with the `organizeDeclarations` rule. When I edit functions with different access controls they will be grouped under the pragma mark comments correct and then I'll make some edits to a function and it will be ungrouped

I've attached a video demonstrating the issue:
https://github.com/user-attachments/assets/abf118de-09ad-4467-a1df-b2a1ee974f82

Here is my `.swiftformat` file:

```
--acronyms ID,URL,UUID
--allman false
--assetliterals visual-width
--beforemarks class,enum,let,struct,typealias,var
--binarygrouping 4,8
--categorymark "MARK: - %c -"
--classthreshold 0
--closingparen balanced
--closurevoid remove
--commas always
--conflictmarkers reject
--decimalgrouping 3,8
--elseposition same-line
--emptybraces no-space
--enumnamespaces always
--enumthreshold 0
--exponentcase lowercase
--exponentgrouping disabled
--extensionacl on-extension
--extensionlength 0
--extensionmark "MARK: - %t + %c"
--fractiongrouping disabled
--fragment false
--funcattributes prev-line
--generictypes 
--groupedextension "MARK: %c"
--guardelse auto
--hexgrouping 4,8
--hexliteralcase uppercase
--ifdef indent
--importgrouping testable-last
--indent 4
--indentcase false
--indentstrings false
--lifecycle layoutSubviews,loadView,viewDidAppear,viewDidDisappear,viewDidLoad,viewWillDisappear,willWillAppear
--lineaftermarks true
--linebreaks lf
--markcategories true
--markextensions always
--marktypes always
--maxwidth 120
--modifierorder override,public,private,lazy,weak,final,convenience
--nevertrailing 
--nospaceoperators 
--nowrapoperators <,>
--octalgrouping 4,8
--operatorfunc no-space
--organizetypes actor,class,enum,struct
--patternlet hoist
--ranges spaced
--redundanttype inferred
--self init-only
--selfrequired 
--semicolons never
--shortoptionals always
--smarttabs enabled
--someAny true
--stripunusedargs closure-only
--structthreshold 0
--tabwidth unspecified
--trailingclosures 
--trimwhitespace always
--typeattributes prev-line
--typeblanklines remove
--typemark "MARK: - %t"
--varattributes same-line
--voidtype void
--wraparguments before-first
--wrapcollections before-first
--wrapconditions preserve
--wrapparameters before-first
--wrapreturntype preserve
--wrapternary default
--wraptypealiases disabled
--xcodeindentation disabled
--yodaswap always
--disable strongOutlets,wrapMultilineStatementBraces
--enable blankLineAfterImports,blankLinesBetweenImports,blockComments,organizeDeclarations,wrapEnumCases,wrapSwitchCases

```

This of course is very irritating and causes a headache in code review as functions are shifted around causing large diffing changes.

This has been reproduced on the latest version `0.54.6`.

**Comment by calda** [comment]
Weird issue, thanks for the report! I'll take a look.

**Comment by calda** [comment]
Here's a fix: #1914

**Comment by cameroncooke** [comment]
Thanks for this I look forward to trying this out.

**Comment by cameroncooke** [comment]
What's the release cadence, will there be a new release soon so I can stop my functions jumping about 🤣

**Comment by nicklockwood** [comment]
There's no particular cadence. Bug fixes are normally released within a few days, but unfortunately due to the massive refactoring recently this next release has been a bit of a struggle to put together because I can't simply cherry pick fixes onto main, so they'll have to be bundled with the next major release.

**Comment by cameroncooke** [comment]
Thanks @nicklockwood

**Comment by nicklockwood** [comment]
@cameroncooke fixed in 0.55.0

**Comment by cameroncooke** [comment]
Thank you @nicklockwood 🙏
